docassemble/EFSPIntegration/efm_client.py

```diff
@@ -8,33 +8,33 @@
 import http.client as http_client
 import logging
 from typing import List
-from docassemble.base.functions import all_variables 
-from docassemble.base.util import log 
+from docassemble.base.functions import all_variables, get_config 
 from docassemble.AssemblyLine.al_document import ALDocumentBundle
 
 class ApiResponse(object):
-  def __init__(self, response_code, error_msg, data):
+  def __init__(self, response_code, error_msg:str, data):
     self.response_code = response_code
     self.error_msg = error_msg
     self.data = data
 
   def __str__(self):
     if self.error_msg:
-      return f'response_code: {self.response_code}, error_msg: {self.error_msg}, data: {data}'
+      return f'response_code: {self.response_code}, error_msg: {self.error_msg}, data: {self.data}'
     else:
       return f'response_code: {self.response_code}, data: {self.data}'
 
 class ProxyConnection(object):
-    def __init__(self, url: str, api_token: str):
+    def __init__(self, 
+                 url:str=get_config('efile proxy').get('url'), 
+                 api_key:str=get_config('efile proxy').get('api key')):
```

Comment:
This looks correct, but there's a weird Python thing with default parameters of class methods as objects--should we make sure that isn't triggered here?

https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments

Reply:
On second thought, I don't think this bug would end up getting triggered, as the param is always gonna be a string, and strings are immutable. But it's a good call to change it anyway.